File: 05_weekday.py
#!/usr/bin/env python3

import datetime
from sys import stdin

__author__      = "Christoph Rist"

# Solve this problem most fast, easy and error-proof
# by making use of built-in standard library functionality.
# ... don't reinvent the wheel ...
 

WEEKDAY_NAMES = {
        0: "Montag",
        1: "Dienstag",
        2: "Mittwoch",
        3: "Donnerstag",
        4: "Freitag",
        5: "Samstag",
        6: "Sonntag"
}

def main():
    lines = stdin.readlines();
    nb_cases = int(lines[0])
    for i in range(1, nb_cases + 1):
        n = lines[i].split(" ")
        yyyy = int(n[0])
        mm = int(n[1])
        dd = int(n[2])
        while yyyy < 1:
            yyyy += 400

        d = datetime.datetime(yyyy, mm, dd)
        print(WEEKDAY_NAMES[d.weekday()])


if __name__ == "__main__":
    main()

File: 05_weekday.cpp
// this is for emacs file handling -*- mode: c++; indent-tabs-mode: nil -*-
// -- BEGIN LICENSE BLOCK ----------------------------------------------
// -- END LICENSE BLOCK ------------------------------------------------
//----------------------------------------------------------------------
/*!\file
 *
 * \author  Christoph Rist <rist.christoph@gmail.com>
 * \date    2016-12-05
 *
 */
//----------------------------------------------------------------------
/***********************************
 * IEEE SB Passau Adventskalender
 * Problem 05
 */

#include <iostream>
#include <string>
#include <ctime>
#include <iomanip>
#include <stdexcept>


std::string getWeekdayString(int wday)
{
  switch (wday)
  {
    case 0: return "Montag";
    case 1: return "Dienstag";
    case 2: return "Mittwoch";
    case 3: return "Donnerstag";
    case 4: return "Freitag";
    case 5: return "Samstag";
    case 6: return "Sonntag";
    default: throw std::runtime_error("Invalid wday");
  }
}

/**
 * Solve this problem most fast, easy and error-proof
 * by making use of standard library functionality.
 * ... don't reinvent the wheel ...
 */

int main(int argc, char* argv[])
{
  uint32_t nb_testcases;
  std::cin >> nb_testcases;

  int y, m, d;
  for (uint32_t i = 0; i < nb_testcases; ++i)
  {
    std::cin >> y >> m >> d;
    std::tm t = {0, 0, 12, d, m - 1, y - 1990, 0, 0, 0};
    std::mktime(&t);
    std::cout <<getWeekdayString(t.tm_wday) <<std::endl;
  }

  return 0;
}
